File: public/actions/javascript/encryptDataLocally.ts
import { CoverCrypt, Policy } from "cloudproof_js";

//
// Encrypt data locally
//
export const encryptDataLocally = async (
  publicKey: Uint8Array,
  policy: Policy,
  accessPolicy: string,
  clearData: Uint8Array
): Promise<Uint8Array> => {
  const { CoverCryptHybridEncryption } = await CoverCrypt();
  const encryption = new CoverCryptHybridEncryption(policy, publicKey);
  return encryption.encrypt(accessPolicy, clearData);
};

File: public/actions/python/encryptDataLocally.py
from cloudproof_py.cover_crypt import CoverCrypt


def encrypt_data_locally( public_key: string, plaintext: string, policy: string, encryption_policy: string):
    """Encrypt data locally"""
    cover_crypt = CoverCrypt()
    ciphertext = cover_crypt.encrypt(
        policy,
        encryption_policy,
        public_key,
        plaintext,
    )
    return ciphertext
